pkgs/development/compilers/ghc/8.6.5-binary.nix

```diff
@@ -24,27 +25,33 @@ let
 in
 
 stdenv.mkDerivation rec {
-  version = "8.6.3";
+  version = "8.6.5";
 
   name = "ghc-${version}-binary";
 
+  # https://downloads.haskell.org/~ghc/8.6.5/
   src = fetchurl ({
     i686-linux = {
-      url = "http://haskell.org/ghc/dist/${version}/ghc-${version}-i386-deb8-linux.tar.xz";
-      sha256 = "0bw8a7fxcbskf93rb4m542ff66vrmx5i5kj77qx37cbhijx70w5m";
+      url = "http://haskell.org/ghc/dist/${version}/ghc-${version}-i386-deb9-linux.tar.xz";
+      sha256 = "1p2h29qghql19ajk755xa0yxkn85slbds8m9n5196ris743vkp8w";
     };
     x86_64-linux = {
-      url = "http://haskell.org/ghc/dist/${version}/ghc-${version}-x86_64-deb8-linux.tar.xz";
-      sha256 = "1m9gaga2pzi2cx5gvasg0rx1dlvr68gmi20l67652kag6xjsa719";
+      url = "http://haskell.org/ghc/dist/${version}/ghc-${version}-x86_64-deb9-linux.tar.xz";
+      sha256 = "1pqlx6rdjs2110g0y1i9f8x18lmdizibjqd15f5xahcz39hgaxdw";
+    };
+    aarch64-linux = {
+      url = "http://haskell.org/ghc/dist/${version}/ghc-${version}-aarch64-ubuntu18.04-linux.tar.xz";
+      sha256 = "11n7l2a36i5vxzzp85la2555q4m34l747g0pnmd81cp46y85hlhq";
     };
     x86_64-darwin = {
       url = "http://haskell.org/ghc/dist/${version}/ghc-${version}-x86_64-apple-darwin.tar.xz";
-      sha256 = "1hbzk57v45176kxcx848p5jn5p1xbp2129ramkbzsk6plyhnkl3r";
+      sha256 = "0s9188vhhgf23q3rjarwhbr524z6h2qga5xaaa2pma03sfqvvhfz";
     };
   }.${stdenv.hostPlatform.system}
     or (throw "cannot bootstrap GHC on this platform"));
 
   nativeBuildInputs = [ perl ];
+  propagatedBuildInputs = lib.optional stdenv.isAarch64 [ llvm ];
```

Comment:
I think the big question from https://github.com/NixOS/nixpkgs/pull/80355#pullrequestreview-364288760 is, "Why is this needed?"

It seems strange that aarch64 would need this, but nothing else would.  Additionally, why is this not needed for the non-bootstrap GHCs?

I don't really have any knowledge of how GHC is built, or how aarch64 works, but this just _seems_ like the wrong solution.

Reply:
I believe I have found the answers.

GHC on non-x86 uses the LLVM codegen by default [citation needed]. When using the LLVM codegen, ghc needs llvm binaries on the path.

> You must install and have LLVM available on your `PATH` for the LLVM code generator to work. Specifically GHC needs to be able to call the `opt` and `llc` tools. 

-- (from [10.10.2. LLVM Code Generator (-fllvm)][ghc -fllvm])

[ghc -fllvm]: https://downloads.haskell.org/~ghc/8.6.5/docs/html/users_guide/codegens.html#llvm-code-generator-fllvm

> It seems strange that aarch64 would need this, but nothing else would. Additionally, why is this not needed for the non-bootstrap GHCs?

It turns out this is required for the non-bootstrap GHCs, and is already present in `propagatedBuildInputs` of, for example, ghc 8.8.3:
https://github.com/NixOS/nixpkgs/blob/bfb747aacf5bccf1609170839d63e519cf63ca0c/pkgs/development/compilers/ghc/8.8.3.nix#L193-L194

I've updated the branch to more closely resemble the non bootstrap GHC structure.